lib/pushmi_pullyu/swift_depositer.rb

```diff
@@ -0,0 +1,55 @@
+require 'openstack'
+
+class PushmiPullyu::SwiftDepositer
+
+  attr_reader :swift_connection, :swift_container
+
+  def initialize(connection, container)
+    raise 'conection can not be nil' if connection.nil?
+    raise 'container can not be nil' if container.nil?
+
+    endpoint     = connection[:endpoint]
+    auth_version = connection[:auth_version]
+
+    @swift_connection = OpenStack::Connection.create(username: connection[:username],
+                                                     api_key: connection[:password],
+                                                     auth_method: 'password',
+                                                     auth_url: "#{endpoint}/auth/#{auth_version}",
+                                                     authtenant_name: connection[:tenant],
+                                                     service_type: 'object-store')
+    @swift_container = container
+    @logger = PushmiPullyu.logger
+  end
+
+  def deposit_file(file_name)
+    # check if file exists
+    raise "File #{file_name} does not exist" unless File.file?(file_name)
+    file_base_name = File.basename(file_name)
+
+    # calculate file hash
+    hash = Digest::MD5.file(file_name).hexdigest
+
+    # get container object
+    era_container = @swift_connection.container(@swift_container)
+
+    # check that container exits
+    raise "Container #{@swift_container} does not exist" if era_container.nil?
```

Comment:
Why would this be `nil`? As noted in a previous review, the `.container` call *raises an exception* if the container doesn't exist. It is impossible for this to be nil given the OpenStack gem's code